app/controllers/balance_transactions_controller.rb

```diff
@@ -0,0 +1,73 @@
+class BalanceTransactionsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :create_new_form, only: %i[new create]
+  before_action :create_edit_form, only: %i[edit update]
+
+  def new
+  end
+
+  def create
+    if @form.validate(create_params)
+      BalanceTransactions::Creator.new(create_params).call
+      flash[:notice] = t('transaction_create')
+    else
+      flash[:alert] = t('transaction_not_create')
+    end
+
+    respond_to do |format|
+      format.html { redirect_to activity_page_path }
+      format.js
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @form.validate(update_params)
+      BalanceTransactions::Updater.new(update_params).call
+      flash[:notice] = t('transaction_update')
+    else
+      flash[:alert] = t('transaction_not_update')
+    end
+
+    respond_to do |format|
+      format.html { redirect_to activity_page_path }
+      format.js
+    end
+  end
+
+  private
+
+  def transaction
+    Transaction.find(params[:id])
+  end
+
+  def balance_transaction
+    transaction.transactinable
+  end
+
+  def create_new_form
+    @form = BalanceTransactionForm.new(BalanceTransaction.new, transactions: Transaction.new)
+  end
+
+  def create_edit_form
+    @form = BalanceTransactionForm.new(balance_transaction, transactions: transaction)
```

Comment:
@marderer we've discussed it before: `#balance_transaction` should go after a call, so, firstly `#create_edit_form` and `#balance_transaction` after it

Reply:
The same thing with `#transaction`